presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -167,6 +157,9 @@ public boolean isFinished()
         if (spilledLookupJoiner.isPresent()) {
             return spilledLookupJoiner.get().isBlocked();
         }
+        if (currentLookupPartition.isPresent()) {
+            return toListenableFuture(currentLookupPartition.get());
```

Comment:
why do you need to block on both `spilledLookupJoiner` and `currentLookupPartition`?